indexer/tasks/src/genesis/genesis_txs.rs

```diff
@@ -128,51 +189,71 @@ async fn handle_txs(
             &Value::new(amount),
         ));
     }
+}
 
-    let inserted_txs = bulk_insert_txs(db_tx, &transactions).await?;
-    let inserted_addresses = Address::insert_many(
-        addresses
-            .iter()
-            .enumerate()
-            .map(|(idx, addr)| addr(inserted_txs[idx].id)),
-    )
-    .exec_many_with_returning(db_tx)
-    .await?;
-
-    let outputs_to_add = inserted_txs
-        .iter()
-        .zip(&inserted_addresses)
-        .enumerate()
-        .map(|(i, (tx, addr))| TransactionOutputActiveModel {
-            address_id: Set(addr.id),
-            tx_id: Set(tx.id),
-            payload: Set(outputs[i].to_bytes()),
-            // recall: genesis txs are hashes of addresses
-            // so all txs have a single output
-            output_index: Set(0),
-            ..Default::default()
-        });
-    let inserted_outputs = TransactionOutput::insert_many(outputs_to_add)
-        .exec_many_with_returning(db_tx)
+// https://github.com/SeaQL/sea-orm/issues/691
+async fn insert_active_models<ActiveModel>(
+    db_tx: &DatabaseTransaction,
+    active_models: Vec<ActiveModel>,
+) -> Result<(), DbErr>
+where
+    ActiveModel: ActiveModelTrait + ActiveModelBehavior + Send + Sync,
+    <<ActiveModel as ActiveModelTrait>::Entity as EntityTrait>::Model: IntoActiveModel<ActiveModel>,
+{
+    // Max parameter count for SQLite: https://www.sqlite.org/limits.html
+    let batch_size = 32766usize
+        / <<ActiveModel as ActiveModelTrait>::Entity as EntityTrait>::Column::iter().count();
+    for chunk in active_models.chunks(batch_size) {
+        ActiveModel::Entity::insert_many(chunk.to_vec())
+            .exec(db_tx)
+            .await?;
+    }
+    Ok(())
+}
+
+async fn insert_active_transaction_models(
+    db_tx: &DatabaseTransaction,
+    active_models: Vec<TransactionActiveModel>,
+    block_id: i32,
+) -> Result<Vec<TransactionModel>, DbErr> {
+    insert_active_models(db_tx, active_models).await?;
+    let models = <entity::prelude::Transaction as EntityTrait>::find()
+        .filter(transaction::Column::BlockId.eq(block_id))
+        .all(db_tx)
         .await?;
+    Ok(models)
+}
 
-    Ok((inserted_txs, inserted_addresses, inserted_outputs))
+async fn insert_active_address_models(
+    db_tx: &DatabaseTransaction,
+    active_models: Vec<AddressActiveModel>,
+    tx_ids: &Vec<i64>,
+) -> Result<Vec<AddressModel>, DbErr> {
+    insert_active_models(db_tx, active_models).await?;
+    let mut all_models = Vec::new();
+    for tx_id in tx_ids {
+        let models = <entity::prelude::Address as EntityTrait>::find()
```

Comment:
This approach will be significantly slower than what we had before. We purposely batch as many things as possible in a single DB request because otherwise the DB overhead gets to be really big

Reply:
IIRC, the reason I was trying to avoid `exec_many_with_returning` was so I could use an in-memory DB, and, with SeaORM, you can only do in-memory with SQLite, which doesn't support returning.

I generally prioritize testibility over performance--I'd rather be confident in my code than have it be fast and wrong. The best solution here would be to decouple Carp from SeaORM entirely, so you can just inject a mock db.

Reply:
Are operations on `db_tx` the same as making a "DB request"? It's the lookups that had to be iterated over here (without returning) but if they are just looking at what's in the local `DatabaseTransaction` rather doing a "DB request" for each, I think that might be fine performance-wise.